File: 0762-prime-number-of-set-bits-in-binary-representation/0762-prime-number-of-set-bits-in-binary-representation.py
class Solution:
    def countPrimeSetBits(self, left: int, right: int) -> int:
        res = 0
        for i in range(left, right + 1):
            if i.bit_count() == 2 or i.bit_count() == 3 or i.bit_count() == 5 or i.bit_count() == 7 or i.bit_count() == 11 or i.bit_count() == 13 or i.bit_count() == 17 or i.bit_count() == 19:
                res += 1
        return res

File: 0762-prime-number-of-set-bits-in-binary-representation/0762-prime-number-of-set-bits-in-binary-representation.cpp
class Solution {
public:
    int countPrimeSetBits(int left, int right) {
        int res = 0;
        for(int i = left; i <= right; i++)
        {
            if(countBits(i) == 2 || countBits(i) == 3 || countBits(i) == 5 || countBits(i) == 7 || countBits(i) == 11 || countBits(i) == 13 || countBits(i) == 17 || countBits(i) == 19 || countBits(i) == 23 || countBits(i) == 29)
            {
                res++;
            }
        }
        return res;
    }
private:
    char countBits(int n)
    {
        char bits = 0;
        int mask = 1;
        for(int i = 0; i < 32; i++)
        {
            if(mask & n)
            {
                bits++;
            }
            mask <<= 1;
        }
        return bits;
    }
};